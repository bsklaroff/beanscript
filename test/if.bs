a = 3
b = 5
c = 6

if a > 2
  print(0)
  if b == 2
    print(3)
  else
    if c != 6
      print(5)
    else if c <= 6
      print(6)
    else
      print(7)
    print(4)
else if a > 1
  print(1)
else
  print(2)

true_print = ->
  print(True)
  return True

false_print = ->
  print(False)
  return False

if not a <= 2
  print(8)

if true_print() and false_print() or false_print() and true_print()
  print(9)

print(10)

if (true_print() or false_print()) and (false_print() or true_print())
  print(11)
