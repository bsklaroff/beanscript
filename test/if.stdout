0
6
4
8
True
False
False
10
True
False
True
11
